.github/workflows/test.yml

```diff
@@ -11,9 +11,49 @@ on:
 
 jobs:
 
-  build:
+  changelog-check:
+    runs-on: ubuntu-20.04
+    if: ${{ github.event_name == 'pull_request' }}
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run check
+      run: |
+        if [ ! -z "${{ github.event.pull_request.number }}" ]; then
+          if echo """${{ github.event.pull_request.body }}""" | grep -qE '^\[no\ changelog\]$'; then
```

Comment:
I think we can escape from escaping hell by using environment variables to pass data into the shell script.

Adding this to the step:

```yaml
env:
  PR_BODY: ${{ github.event.pull_request.body }}
```

should allow us to simplify this line to:

```bash
if grep -qE '^\[no\ changelog\]$' <<<"$PR_BODY"; then
```

(Whether we use `echo ... | grep` or a here-string `<<<WORD` is irrelevant though)

Reply:
That's realy better. My first intention was to use a label instead of the PR body but the labels can only be added by members and not by contributors.